### PR TITLE
add depsdev scanner implementation

### PR DESCRIPTION
**Pull request by lumjjb**
# Description of the PR

Implement deps.dev scanner on ingestion using depsdevclient introduced in #2278 

# PR Checklist

- [ ] All commits have [a Developer Certificate of Origin (DCO)](https://wiki.linuxfoundation.org/dco) -- they are generated using `-s` flag to `git commit`.
- [ ] All new changes are covered by tests
- [ ] If GraphQL schema is changed, `make generate` has been run
- [ ] If GraphQL schema is changed, GraphQL client updates/additions have been made
- [ ] If OpenAPI spec is changed, `make generate` has been run
- [ ] If ent schema is changed, `make generate` has been run
- [ ] If `collectsub` protobuf has been changed, `make proto` has been run
- [ ] All CI checks are passing (tests and formatting)
- [ ] All dependent PRs have already been merged

### CODE REVIEW

File: pkg/ingestor/parser/common/scanner/scanner.go [hunk 1]
```diff
@@ -115,7 +118,64 @@ func purlsLicenseScanWithClient(
 }
 
 func PurlsDepsDevScan(ctx context.Context, purls []string) ([]assembler.CertifyScorecardIngest, []assembler.HasSourceAtIngest, error) {
```
Comment (pxp928) [inline pkg/ingestor/parser/common/scanner/scanner.go:120]: looks like this is set to false as its not part of the flags yet? By default it should be false but there is no flag to enable it.

File: pkg/ingestor/parser/common/scanner/scanner.go [hunk 2]
```diff
@@ -115,7 +118,64 @@ func purlsLicenseScanWithClient(
...
-	return nil, nil, fmt.Errorf("Unimplemented")
+	certifyScorecards := []assembler.CertifyScorecardIngest{}
+	hasSourceAts := []assembler.HasSourceAtIngest{}
+
+	ddc, err := deps_dev_client.NewDepsClient(ctx)
```
Comment (pxp928) [inline pkg/ingestor/parser/common/scanner/scanner.go:124]: What is the added time to the ingestion when this is enabled? Can you post in a comment the time with this enabled and without?

**Review by pxp928 — APPROVED** [review]
Overall looks good but missing the CLI flag.

**Review by mihaimaruseac — APPROVED** [review]

File: pkg/ingestor/parser/common/scanner/scanner.go (hunk 1 above)
Comment (lumjjb) [reply pkg/ingestor/parser/common/scanner/scanner.go:120]: I think this should be implemented in : https://github.com/guacsec/guac/pull/2265/files#diff-fc14c5bdd69817e3f9b1f331130f5055f448774eb64db13c0ef23a8e882cd1d4R74-R76? Does that address your comment?

Comment (pxp928) [reply pkg/ingestor/parser/common/scanner/scanner.go:120]: did we need to add it here: https://github.com/guacsec/guac/blob/main/pkg/cli/store.go

File: pkg/ingestor/parser/common/scanner/scanner.go (hunk 2 above)
Comment (lumjjb) [reply pkg/ingestor/parser/common/scanner/scanner.go:127]: Added a comment here that each scan takes about 300ms - 500ms ranging from 1-30 PURLs.

**Pull request merged**